src/metabase/api/email.clj

```diff
@@ -73,20 +73,27 @@
   [:as {settings :body}]
   {settings su/Map}
   (validation/check-has-application-permission :setting)
-  (let [settings (-> settings
-                     (select-keys (keys mb-to-smtp-settings))
-                     (set/rename-keys mb-to-smtp-settings))
-        settings (cond-> settings
-                   (string? (:port settings))     (update :port #(Long/parseLong ^String %))
-                   (string? (:security settings)) (update :security keyword))
+  ;; the frontend has access to an obfuscated version of the password. Watch for whether it sent us a new password or
+  ;; the obfuscated version
+  (let [obfuscated? (and (:email-smtp-password settings) (email/email-smtp-password)
+                         (= (:email-smtp-password settings) (setting/obfuscate-value (email/email-smtp-password))))
```

Comment:
don't we allow * in password itself? what if obfuscated password is equal to the ordinary password ceteris paribus?

for example, someone makes bad decisions in life and sets their ordinary smtp password as "************************sdfwefwfsdfwef" or whatever

Reply:
That would be fine actually. Because 
```clojure
setting=> (obfuscate-value "************************sdfwefwfsdfwef")
"**********ef"
```

So if the payload contained `"**********ef"` we would recognize this is the obfuscated value and replace it with `"************************sdfwefwfsdfwef"`. I think the only place where this would fail is if someone has a password and wants to change it to the same value as the obfuscated password. ie password is `"password"`, and then the change it to `"**********rd"`. We would consider the newly supplied password as just the obfuscated password and continue to substitute the original `"password"` and they would never be able to change it. I think this is a very small likelihood of happening.